index.d.ts

```diff
@@ -0,0 +1,110 @@
+import { Action, ActionCreator, AnyAction, StoreEnhancer, Store } from 'redux';
+
+export interface StoreCreator {
+  <S, A extends Action>(
+    reducer: LoopReducer<S, A>,
+    preloadedState: S,
+    enhancer: StoreEnhancer<S>
+  ): Store<S>;
+}
+
+export type Loop<S, A extends Action> = [S, CmdType<A>];
+
+export interface LoopReducer<S, A extends Action> {
+  (state: S | undefined, action: A): S | Loop<S, A>;
+}
+
+declare interface LiftedLoopReducer<S, A extends Action> {
```

Comment:
why is this declared but LoopReducer is exported?

Reply:
ah yeah this should probably be exported too